Streetcode/Streetcode.XUnitTest/MediatRTests/Media/Audio/GetAudioByStreetcodeIdTests.cs

```diff
@@ -0,0 +1,98 @@
+﻿using Xunit;
+using Moq;
+using AutoMapper;
+using Streetcode.DAL.Repositories.Interfaces.Base;
+using Streetcode.BLL.MediatR.Media.Audio.GetByStreetcodeId;
+using Streetcode.BLL.DTO.Media;
+using Microsoft.EntityFrameworkCore.Query;
+using System.Linq.Expressions;
+
+using Entities = Streetcode.DAL.Entities.Media;
+
+namespace Streetcode.XUnitTest.MediatRTests.Media.Audio
+{
+
+    public class GetAudioByStreetcodeIdTests
+    {
+        [Theory]
+        [InlineData(1)]
+        public async Task GetAudioByStreetcodeId_ExistingId(int id)
+        {
+            var repository = new Mock<IRepositoryWrapper>();
+            var mockMapper = new Mock<IMapper>();
+
+            var testAudio = new Entities.Audio() { StreetcodeId = id };
+
+            repository.Setup(repo => repo.AudioRepository
+            .GetFirstOrDefaultAsync(It.IsAny<Expression<Func<Entities.Audio, bool>>>(), It.IsAny<Func<IQueryable<Entities.Audio>, IIncludableQueryable<Entities.Audio, Entities.Audio>>?>()))
+            .ReturnsAsync(testAudio);
+
+            mockMapper.Setup(x => x.Map<AudioDTO>(It.IsAny<Entities.Audio>()))
+            .Returns((Entities.Audio sourceAudio) =>
+            {
+                return new AudioDTO { StreetcodeId = sourceAudio.StreetcodeId };
+            });
+            // arrange
+            var handler = new GetAudioByStreetcodeIdQueryHandler(repository.Object, mockMapper.Object);
```

Comment:
this all is arrange, but this creation of mocks can be moved to separate methods

Reply:
created a constructor for this, is it okay, or using method is a better approach?